### ISSUE TITLE
Code for training

### ISSUE THREAD

**Issue by peteryanj** [mattpoggi/pydnet#1]
Dear. mattpoggi
  Recently I read your paper "Towards real-time unsupervised monocular depth estimation on CPU" . Your clear and compact declaration inspired me a lot and I think it is very interesting.But I got some trouble when I train this code,Could you please provide the source code of training code?
  Thank you very much! 
  
Best regards

**Comment by mattpoggi** [comment]
Hi @peteryanj ,
thanks for your appreciation. I plan to upload the training code (hopefully, before IROS), by the way it is mostly a fork from Clément Godard repository https://github.com/mrharicot/monodepth that already contains amazing tools for training. Actually, you can plug-in pydnet model into his framework and train it as he does with vgg and resnet models.

**Comment by peteryanj** [comment]
Dear. mattpoggi
   Thank you for your email, I will try it by using  https://github.com/mrharicot/monodepth and hope you will upload the training code ,thanks very much!
   Best regards

**Comment by saetlan** [comment]
Hello @mattpoggi, thanks for this great contribution. Do you still plan to upload the training code ?

**Comment by mattpoggi** [comment]
Dear @saetlan (and @peteryanj as well),
thanks for pointing it out, I totally forgot to upload it (sorry about that). If you pull the last version, you will find the "training_code" folder containing some scripts. You can take and plug them inside monodepth framework https://github.com/mrharicot/monodepth (you have to replace the original "monodepth_model.py" script). Then, you can run monodepth_main.py as is and it will train pydnet. This is what I did to train the networks reported in the paper so it should work for you as well, otherwise, do not hesitate to reply here. Sorry again for late and thanks for reminding me to upload the code.
Best

**Comment by mattpoggi** [comment]
@saetlan Anyway, I uploaded a script (experiments.py) that you can use to reproduce the results on the Eigen split. As filenames list, you can use the same from Godard's repository.
I hope this helps

**Comment by saetlan** [comment]
@mattpoggi I checked again and the numbers are correct. I thought the provided checkpoint was trained on Kitti only. My bad :) Thanks again !

**Comment by songya** [comment]
Dear @mattpoggi ,
I would like to ask you about get_disp function.
In the Godard's paper, the code is 
disp = 0.3 * self.conv(x, 2, 3, 1, tf.nn.sigmoid)
But yours is disp = 0.3 * tf.nn.sigmoid(tf.slice(x, [0,0,0,0], [-1,-1,-1,2]))
My question is why you change get_disp function. 
Did the function improve the performance?

**Comment by mattpoggi** [comment]
Dear @songya ,
the main reason is to speed up the network and to keep compatibility with another framework requiring minimum 8 channels as output for 2Dconv.
I also converted elu to leaky rely for the sake of speed, being the exponential slower.
Best

**Comment by songya** [comment]
Dear @mattpoggi , 
Thank you for the quick reply.
You mentioned that the framework required minimum 8 channels as output for 2D conv.
Could you explain the reason in details?
In my case, I could reduce the channel size into 4 although the results were not so good though.

**Comment by mattpoggi** [comment]
It's a third-party framework, I can't say more. I think it is mainly to exploit parallel computing.

**Comment by emad555** [comment]
I am trying to get the results using the following command:

python3 experiments.py --datapath PATH_TO_KITTI --filenames PATH_TO_FILELIST --checkpoint_dir checkpoint/IROS18/pydnet --resolution 1

what should I replace:
PATH_TO_KITTI
PATH_TO_FILELIST

i used the path of KITTI path in my pc but it is not working.
